back-end/src/main/java/com/thesis/projectopportunities/controller/PositionController.java

```diff
@@ -74,13 +75,21 @@ public ResponseEntity<PaginatedModel<PositionDto>> getPositions(@RequestParam(re
 	}
 
 	@DeleteMapping("/positions/{id}")
-	public void deletePosition(@PathVariable int id) {
+	public void deletePosition(@PathVariable int id, Authentication authentication) {
+
+		var position =
+			positionRepo.findById(id);
+
+		if (position.isEmpty() || !companyPermissionRepo.existsByCompanyIdAndUsername(position.get().getCompany().getId(),
+			authentication.getName())) {
+			return;
+		}
 		positionRepo.deleteById(id);
 	}
```

Comment:
You should not have a repository in a controller if you have a service. You can call a service method and then the service should call the repository. This is how it should work everywhere